evap/staff/templates/staff_semester_view.html

```diff
@@ -292,50 +359,49 @@ <h3>
-                                    data-toggle="tooltip" data-placement="top" title="{% trans 'Text answers awaiting review' %}"></a>
-                                <a role="button" class="btn btn-sm btn-light fas fa-comment icon-green" data-filter="fa-comment icon-green"
-                                    data-toggle="tooltip" data-placement="top" title="{% trans 'Text answers reviewed' %}"></a>
-                                <a role="button" class="ml-2 btn btn-sm btn-light fas fa-chart-bar icon-gray" data-filter="fa-chart-bar icon-gray"
-                                    data-toggle="tooltip" data-placement="top" title="{% trans 'Evaluation not finished' %}"></a>
-                                <a role="button" class="btn btn-sm btn-light fas fa-chart-bar icon-red" data-filter="fa-chart-bar icon-red"
-                                    data-toggle="tooltip" data-placement="top" title="{% trans 'Results not yet published although they probably could be' %}"></a>
-                                <a role="button" class="btn btn-sm btn-light fas fa-chart-bar icon-yellow" data-filter="fa-chart-bar icon-yellow"
-                                    data-toggle="tooltip" data-placement="top" title="{% trans 'Results not yet published' %}"></a>
-                                <a role="button" class="btn btn-sm btn-light fas fa-chart-bar icon-green" data-filter="fa-chart-bar icon-green"
-                                    data-toggle="tooltip" data-placement="top" title="{% trans 'Results published' %}"></a>
-                            </div>
-                        </div>`
-                    );
-                    $('[data-filter]').click( function(e) {
-                        $(e.target).toggleClass("active");
-                        $("[data-filter].active").not(e.target).removeClass("active");
+                    $('#course-filter-buttons [data-filter]').click(function(e) {
+                        // toggle active states and refresh table
+                        var button = $(e.target).closest("a");
+                        button.toggleClass("active");
+                        $("[data-filter].active").not(button).removeClass("active");
                         table.draw();
-                    } );
-                    activateTooltips('.filter-toolbar');
-                },
+
+                        // store which button was selected
+                        localStorage.removeItem('course-filter_{{ semester.id }}');
+                        if($("[data-filter].active").length)
+                        {
+                            localStorage['course-filter_{{ semester.id }}'] = $("[data-filter].active")[0].dataset.filter;
+                        }
+                    });
+                    // helper function to count number of lines with certain icon
+                    countIcons = function(obj) {
+                        var count = 0;
+                        $.each([1, 2, 3, 4], function(index, value) {
```

Comment:
you could use forEach and the arrow notation for something like this:
```
[1,2,3,4].forEach( i => { 
    count += table.column(i).data().filter( (value, index) => value.includes(obj.dataset.filter)).count();
}
```

okay fancy one-liner incoming
`return [1,2,3,4].reduce( (sum, i) => sum + table.column(i).data().filter( (value, index) => value.includes(obj.dataset.filter)).count(), 0);`

:) but yeah, that probably doesn't help anymore...

in any case, i would inline the function (and maybe call the variable iconCount).